app/models/user.rb

```diff
@@ -84,6 +91,22 @@ class User < ApplicationRecord
 
   scope :alertable, -> { where(role: [User.roles[:staff], User.roles[:school_admin], User.roles[:volunteer]]) }
 
+  scope :mailchimp_update_required, -> do
```

Comment:
Whilst I like the light approach on the callback side - definitely good that no extra database writes are done there but I do worry that this query is rather large. Assume it is something that would be run in a batch job in the middle of the night?

Reply:
Planning to run daily initially, but maybe more frequently if needed.

I'm not too worried about it as in my db (with statuses set on ~1300 users, but no updates pushed) it runs in 28ms. The query is mostly LEFT JOINS which should be optimised well in the database.

The query is not fetching anything from the joined tables, so doesn't need to pull anything back from the database other than the User attributes. 

Can see how it runs in practice though.

Reply:
That sounds good then 👍